metacat-common-server/src/main/java/com/netflix/metacat/common/server/connectors/ValidatingConnectorCatalogService.java

```diff
@@ -18,73 +22,119 @@
 import java.util.List;
 
 /**
- * Connector that throttles calls to the connector based on the contextual request name
- * and the resource. Not all APIs can be throttles since we may not have a resource
- * but those are a small monitory
+ * Connector that validates requests based on the request context and/or resource.
  */
 @Slf4j
 @RequiredArgsConstructor
-public class ThrottlingConnectorCatalogService implements ConnectorCatalogService {
+public class ValidatingConnectorCatalogService implements ConnectorCatalogService {
     @Getter
     @NonNull
     private final ConnectorCatalogService delegate;
     @NonNull
     private final RateLimiter rateLimiter;
+    private final boolean rateLimiterEnabled;
+    @NonNull
+    private final Authorization authorization;
+    private final boolean authorizationEnabled;
 
     @Override
     public void create(final ConnectorRequestContext context, final CatalogInfo resource) {
-        checkThrottling(MetacatContextManager.getContext().getRequestName(), resource.getName());
+        if (rateLimiterEnabled) {
+            checkThrottling(MetacatContextManager.getContext().getRequestName(), resource.getName());
+        }
+        if (authorizationEnabled) {
+            authorize(MetacatContextManager.getContext(), resource.getName());
+        }
         delegate.create(context, resource);
     }
 
     @Override
     public void update(final ConnectorRequestContext context, final CatalogInfo resource) {
-        checkThrottling(MetacatContextManager.getContext().getRequestName(), resource.getName());
+        if (rateLimiterEnabled) {
+            checkThrottling(MetacatContextManager.getContext().getRequestName(), resource.getName());
+        }
+        if (authorizationEnabled) {
+            authorize(MetacatContextManager.getContext(), resource.getName());
+        }
         delegate.update(context, resource);
     }
 
     @Override
     public void delete(final ConnectorRequestContext context, final QualifiedName name) {
-        checkThrottling(MetacatContextManager.getContext().getRequestName(), name);
+        if (rateLimiterEnabled) {
+            checkThrottling(MetacatContextManager.getContext().getRequestName(), name);
+        }
+        if (authorizationEnabled) {
```

Comment:
The responsibility for this and similar `if` blocks should lie with the Authorization implementation, rather than the caller performing the checks.